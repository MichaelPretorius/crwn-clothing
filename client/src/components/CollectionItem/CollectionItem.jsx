import React from 'react';
import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cartActions';
import {
  CollectionItemContainer,
  CollectionFooterContainer,
  BackgroundImage,
  NameContainer,
  PriceContainer,
  ButtonContainer
} from './CollectionItem.styles';

const CollectionItem = ({ item, addItem }) => {
  const { name, imageUrl, price } = item;

  return (
    <CollectionItemContainer>
      <BackgroundImage className="image" imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>R{price}</PriceContainer>
      </CollectionFooterContainer>
      <ButtonContainer inverted onClick={() => addItem(item)}>
        Add To Cart
            </ButtonContainer>
    </CollectionItemContainer>
  );
}

export default connect(null, { addItem })(CollectionItem);