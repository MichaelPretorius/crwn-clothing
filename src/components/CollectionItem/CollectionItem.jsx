import React from 'react';
import { connect } from 'react-redux';

import './CollectionItem.scss';
import CustomButton from '../CustomButton/CustomButton';
import { addItem } from '../../redux/cart/cartActions';

const CollectionItem = ({ item, addItem }) => {
    const { name, imageUrl, price } = item;

    return (
        <div className="collection-item">
            <div className="image" style={{
                backgroundImage: `url(${imageUrl})`
            }} />
            <div className="collection-footer">
                <span className="name">{name}</span>
                <span className="price">R{price}</span>
            </div>
            <CustomButton inverted onClick={() => addItem(item)}>
                Add To Cart
            </CustomButton>
        </div>
    );
}

export default connect(null, { addItem })(CollectionItem);