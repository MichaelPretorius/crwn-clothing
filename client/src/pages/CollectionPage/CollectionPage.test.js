import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPage } from './CollectionPage';
import CollectionItem from '../../components/CollectionItem/CollectionItem';

describe('CollectionPage', () => {
  let wrapper;
  let mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }];
  beforeEach(() => {
    const mockCollection = {
      items: mockItems,
      title: 'Test'
    };
  
    wrapper = shallow(<CollectionPage collection={mockCollection} />);
  });
  
  test('should render CollectionPage component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  })

  test('should render same number of CollectionItems as collection array', () => {
    expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
  })
})

