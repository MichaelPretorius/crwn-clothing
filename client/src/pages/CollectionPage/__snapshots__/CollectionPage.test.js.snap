// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage should render CollectionPage component 1`] = `
"<styled.div>
  <styled.h2>
    Test
  </styled.h2>
  <CollectionItemsContainer>
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
  </CollectionItemsContainer>
</styled.div>"
`;
